### PR TITLE
Convert Lego Star Wars fix into hack method

### PR DESCRIPTION
**Pull request by RadWolfie**
Titles tested as working:
* Crimson Skies
* FIFA 2002 World Cup (**fixed**)
* Gauntlet: Dark Legacy
* Ghost Recon
* Ghost Recon 2
* Ghost Recon: Island Thunder
* Lego Star Wars
* Lego Star Wars 2
* Mafia
* Red Faction 2
* Roller Coaster Tycoon
* Smashing Drive
* Turok Evolution (**fixed**)

### CODE REVIEW

File: src/CxbxKrnl/EmuXapi.cpp [hunk 1]
```diff
@@ -344,6 +344,40 @@ bool TitleIsJSRF()
...
+	static bool result = false;
+
+	// Prevent running the check every time this function is called
+	if (detected) {
+		return result;
+	}
+
+	// Array of known Lego Star Wars title IDs, must be 0 terminated
+	DWORD titleIds[] = {
+		0x4553001D, // NTSC
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuXapi.cpp]: Is this another version that would need to be added to this list? : http://redump.org/disc/24495/

**Comment by PatrickvL** [comment]
Looking good

File: src/CxbxKrnl/EmuXapi.cpp (hunk 1 above)
Comment (RadWolfie) [reply src/CxbxKrnl/EmuXapi.cpp]: ES-029 is the same as 0x4553001D hex format. Although, v1.01 is region free?

**Comment by gandalfthewhite19890404** [comment]
2002 World Cup - confirmed, works.
image

**Comment by RadWolfie** [comment]
Resolved #1315

**Comment by LukeUsher** [comment]
I don't like adding more hacks, but i'll include this: The HLE Input hopefully won't be around much longer with @ergo720's OHCI work going on

**Comment by RadWolfie** [comment]
No input Lego Star Wars fix is a hack. What I do is convert the "hack" fix into a hack for specific title id (Lego Star Wars). Therefore, nothing is added as hack. 😋

Beside, the hack is simulating the way pull request #1300 does in one area instead of two areas. Always insertion, return true, and no removal.

**Pull request merged**